### PR TITLE
Fix - Ui/Ux

### LINKED ISSUE #75: fix UI issues

### PR DESCRIPTION
**Pull request by aanish27**
fixed #75 @Raza9798  please review

### CODE REVIEW

**Review by Raza9798 — CHANGES_REQUESTED** [review]
image

place the title in the search bar and remove the search bar

**Review by Raza9798 — CHANGES_REQUESTED** [review]
https://github.com/aanish27/employee-profile-manager/issues/75#issuecomment-2426852913

https://github.com/aanish27/employee-profile-manager/issues/75#issuecomment-2426855255

**Comment by aanish27** [comment]
> [#75 (comment)](https://github.com/aanish27/employee-profile-manager/issues/75#issuecomment-2426852913)
> 
> [#75 (comment)](https://github.com/aanish27/employee-profile-manager/issues/75#issuecomment-2426855255)

Requested Changes are done Please review again @Raza9798

**Review by Raza9798 — CHANGES_REQUESTED** [review]
https://github.com/aanish27/employee-profile-manager/issues/75#issuecomment-2427363449

**Comment by aanish27** [comment]
> Requested Changes are done Please review again @Raza9798

Requested Change is done Please review again @Raza9798 #75

**Comment by aanish27** [comment]
#75 Mentioned issues are resolved @Raza9798

**Review by Raza9798 — CHANGES_REQUESTED** [review]
https://github.com/aanish27/employee-profile-manager/issues/75#issuecomment-2429692674

**Review by Raza9798 — CHANGES_REQUESTED** [review]
https://github.com/aanish27/employee-profile-manager/issues/75#issuecomment-2429799294

File: resources/views/components/nav-bar.blade.php [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
-        </form>
+        <h1 class="fs-5" >{{ $title }}</h1>
     </div>
 
     <div class="nav-right d-flex align-items-center gap-2 ">
-        <svg xmlns="http://www.w3.org/2000/svg" width="25" height="25" fill="#022658" class="bi bi-bell" viewBox="0 0 16 16">
-            <path d="M8 16a2 2 0 0 0 2-2H6a2 2 0 0 0 2 2M8 1.918l-.797.161A4 4 0 0 0 4 6c0 .628-.134 2.197-.459 3.742-.16.767-.376 1.566-.663 2.258h10.244c-.287-.692-.502-1.49-.663-2.258C12.134 8.197 12 6.628 12 6a4 4 0 0 0-3.203-3.92zM14.22 12c.223.447.481.801.78 1H1c.299-.199.557-.553.78-1C2.68 10.2 3 6.88 3 6c0-2.42 1.72-4.44 4.005-4.901a1 1 0 1 1 1.99 0A5 5 0 0 1 13 6c0 .88.32 4.2 1.22 6"/>
-        </svg>
+        <a href="{{ route('logout') }}" onclick="event.preventDefault(); document.getElementById('logout-form').submit();" class="rounded-5 border-black border-1" title="Logout">
+            <svg xmlns="http://www.w3.org/2000/svg" width="22" height="22" fill="#022658" class="bi bi-box-arrow-right" viewBox="0 0 16 16">
```
Comment (Raza9798) [inline resources/views/components/nav-bar.blade.php:13]: For icons use `webfont` icons, instead of SVG icons. replace all SVG icons into Webfont icons

**Review by Raza9798 — CHANGES_REQUESTED** [review]

File: resources/views/admin.blade.php [hunk 2]
```diff
@@ -367,6 +408,13 @@ function storeEmployee(){
                     $('#company-branch').val($(this).find(':selected').data('branch'))
                 });
 
+                $('#dt-length-0').change(function() {
```
Comment (Raza9798) [inline resources/views/admin.blade.php:411]: Please explain this code

**Review by Raza9798 — CHANGES_REQUESTED** [review]

**Review by Raza9798 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/EmployeeController.php [hunk 3]
```diff
@@ -11,52 +11,88 @@
...
-                                  ->get();
-
-          $response = [
-              'draw' => intval($draw),
-              'recordsTotal' => intval($totalEmployees),
-              'recordsFiltered' => $filteredEmployees,
-              'data' => $employees
-          ];
-          return Response::json($response);
+        //SERVER SIDE RENDERING HANDLE FOR data table
```
Comment (Raza9798) [inline app/Http/Controllers/EmployeeController.php:26]: I noticed you have implemented the `order` query on collection it should works on MySQL not in array.

**Review by Raza9798 — CHANGES_REQUESTED** [review]

File: resources/views/admin.blade.php (hunk 2 above)
Comment (aanish27) [reply resources/views/admin.blade.php:411]: the above code handles the misalignment cause the appearance of scroll bar by changing the width the header... @Raza9798 

image

File: resources/views/admin.blade.php [hunk 4]
```diff
@@ -6,15 +6,14 @@
         <meta name="csrf-token" content="{{ csrf_token() }}">
         <title>Employee Manager</title>
 
-
         @vite([ 'resources/css/app.css', 'resources/js/app.js',  'resources/css/custom.data-table.css' ])
-
+        <link rel="stylesheet" href="https://cdn.datatables.net/2.1.8/css/dataTables.bootstrap5.css">
```
[the remark this answers is not in the document]
Comment (aanish27) [reply resources/views/admin.blade.php:10]: Got it Removed @Raza9798

File: app/Http/Controllers/EmployeeController.php (hunk 3 above)
Comment (aanish27) [reply app/Http/Controllers/EmployeeController.php:26]: sorry couldn't understand Can you make it more clear? @Raza9798

File: resources/views/admin.blade.php [hunk 5]
```diff
@@ -186,6 +186,17 @@
                                     return;
                             }
                         });
+
+                        let btn = $('<button class="bi bi-filter btn btn-outline-secondary rounded-1 px-1 py-1 mx-0" title="Clear Filter">Clear</button>')
```
Comment (Raza9798) [inline resources/views/admin.blade.php:190]: `https://datatables.net/reference/api/columns.adjust()`
Refer this code to adjust the column

Comment (aanish27) [reply resources/views/admin.blade.php:190]: > https://datatables.net/reference/api/columns.adjust()

Tried this and it didn't work and found that the misalignment is caused by fixed width set by the custom CSS which is needed for the table to fit on view port when the sidebar is expanded...therefore changing the only solution is to remove the fixed width @Raza9798  #76

File: resources/css/custom.data-table.css [hunk 6]
```diff
@@ -1,58 +1,28 @@
-#btn-add-record{
```
Comment (Raza9798) [inline resources/css/custom.data-table.css:1]: remove `resources/css/custom.data-table.css` file and use inbuild styling for datatable

**Review by Raza9798 — CHANGES_REQUESTED** [review]

File: resources/views/admin.blade.php (hunk 5 above)
Comment (Raza9798) [reply resources/views/admin.blade.php:190]: https://github.com/aanish27/employee-profile-manager/pull/76#discussion_r1815542983

File: app/Http/Controllers/CompanyController.php [hunk 7]
```diff
@@ -21,16 +22,39 @@ public function draw(Request $request){
         $start = $request->query('start', 0);
         $length = $request->query('length', 10);
         $totalCompanys =   Company::count();
+        $arrayColumns = ['', 'name', 'branch','country', 'address', 'employees_count', 'projects_count'];
```
Comment (Raza9798) [inline app/Http/Controllers/CompanyController.php:25]: remove multiple `MODEL` calls and improve the search in Company

File: app/Http/Controllers/EmployeeController.php [hunk 8]
```diff
@@ -11,52 +11,88 @@
...
+        $search = $request->query('search')['value'];
+        $draw = $request->query('draw', 1);
+        $start = $request->query('start', 0);
+        $length = $request->query('length', 10);
+        $totalEmployees =   Employee::count();
+        $arrayColumns = ['','company.name', 'company.branch', 'name', 'position', 'dob', 'email', 'phone', 'address', 'bankAccount.account_no'];
+        $arr = array();
+        if(is_null($search)){
```
Comment (Raza9798) [inline app/Http/Controllers/EmployeeController.php:34]: remove multiple `MODEL` calls and improve the search in Employeees

**Review by Raza9798 — CHANGES_REQUESTED** [review]

**Pull request merged**